src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -812,8 +832,164 @@ def __set_errors_json(self, error_count_by_operation, errors_by_operation):
             "details": errors_by_operation,
             "message": message
         }
+
+    def __set_error_detail(self, error_code, message):
+        formatted_message = self.__ensure_error_message_restriction_compliance(message)
+        return {
+            "code": str(error_code),
+            "message": str(formatted_message)
+        }
     # endregion
 
+    # region - Patch Truncation
+    def log_truncated_packages(self):
+        """ log the removed packages from patches in CoreMain after main operation are marked completed """
+        if not len(self.__assessment_packages_removed) == 0:
+            self.composite_logger.log_debug("Total number of packages removed from assessment packages list is: {0}, ".format(len(self.__assessment_packages_removed)))
+            self.composite_logger.log_debug("Packages removed from assessment packages list: {0}, ".format(self.__assessment_packages_removed))
+
+        if not len(self.__installation_packages_removed) == 0:
+            self.composite_logger.log_debug("Total number of packages removed from installation packages list is: {0}, ".format(len(self.__installation_packages_removed)))
+            self.composite_logger.log_debug("Packages removed from installation packages list: {0}".format(self.__installation_packages_removed))
```

Comment:
Same question: Aren't both of these logs saying the exact same thing?
General ask: Please self-review code before other reviewers take a look.

Reply:
combine the two into one, done